src/lm.jl

```diff
@@ -156,7 +156,32 @@ function coeftable(mm::LinearModel)
               ["x$i" for i = 1:size(mm.pp.X, 2)], 4)
 end
 
-predict(mm::LinearModel, newx::Matrix) =  newx * coef(mm)
+predict(mm::LinearModel, newx::AbstractMatrix) = newx * coef(mm)
+
+"""
+For linear models, specifying `interval_type` will return a 3-tuple with the
+predicted values, the lower and the higher confidence bound. Confidence intervals
+delimit the uncertainty of the estimate of the predicted values, prediction
+intervals delimit the estimated bounds for any new data points from the
+population.
+
+# Arguments
+* `interval_type::Symbol`: one of `:confint` or `:predint`
+* `level=0.95`: the level of the confidence interval
+* `alpha=1-level`: an alternative way of specifying the level of the interval
+"""
+function predict(mm::LinearModel, newx::AbstractMatrix, interval_type::Symbol; level = 0.95, alpha = 1 - level)
+    retmean = newx * coef(mm)
+    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented
+
+    R = cholfact!(mm.pp)[:U] #get the R matrix from the QR factorization
+    residvar = (ones(size(newx,2),1) * deviance(mm)/dof_residual(mm))
+    retvariance = (newx/R).^2 * residvar
+
+    interval = quantile(TDist(dof_residual(mm)), alpha/2) * sqrt.(retvariance)
+    retmean, retmean .+ interval, retmean .- interval
```

Comment:
I find a matrix more natural, since it allows you to extract a row easily. As Andreas noted, if you want only one prediction, the function could avoid allocating by returning a tuple of scalars. But I think this should be a different method which would dispatch on `newX::AbstractVector` instead. It could be added if it turns out to be useful.